core/src/main/scala/org/apache/spark/sql/crossdata/catalog/CatalogChain.scala

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright (C) 2015 Stratio (http://stratio.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.spark.sql.crossdata.catalog
+
+import org.apache.spark.sql.catalyst.plans.logical.LogicalPlan
+import org.apache.spark.sql.catalyst.{CatalystConf, TableIdentifier}
+import org.apache.spark.sql.crossdata.XDContext
+import org.apache.spark.sql.crossdata.catalog.XDCatalog.{CrossdataTable, ViewIdentifier}
+
+import scala.collection.SeqView
+
+/*
+  Write through (always true for this class)-> Each write is synchronously done to all catalogs in the chain
+  No-Write allocate (always true) -> A miss at levels 0...i-1,i isn't written to these levels when found at level i+1
+ */
+class CatalogChain(
```

Comment:
Nice! It is similar to how SparkSession catalog will be.  I'd use explicit interfaces to make it easier to implement specific catalogs.